src/manifolds/SymplecticGrassmann.jl

```diff
@@ -0,0 +1,120 @@
+@doc raw"""
+    SymplecticGrassmann{T,𝔽} <: AbstractEmbeddedManifold{𝔽, DefaultIsometricEmbeddingType}
+
+The symplectic Grassmann manifold consists of all symplectic subspaces of
+``ℝ^{2n}`` of dimension ``2k``, ``n ≥ k``.
+
+Points on this manifold can be represented as corresponding representers on the [`SymplecticStiefel`](@ref)
+
+```math
+\operatorname{SpGr}(2n,2k) = \bigl\{ \operatorname{span}(p)\ \big| \ p ∈ \operatorname{SpSt}(2n, 2k, ℝ)\},
+```
+
+or as projectors
+
+```math
+\operatorname{SpGr}(2n, 2k, ℝ) = \bigl\{ p ∈ ℝ^{2n×2n} \ \big| \ p^2 = p, \operatorname{rank}(p) = 2k, p^+=p \bigr\},
+```
+
+where ``⋅^+`` is the [`symplectic_inverse`](@ref).
+See also [`ProjectorPoint`](@ref) and [`StiefelPoint`](@ref) for these two representations,
+where arrays are interpreted as those on the Stiefel manifold.
+
+With respect to the quotient structure, the canonical projection ``π = π_{\mathrm{SpSt},\mathrm{SpGr}}`` is given by
+
+```math
+π: \mathrm{SpSt}(2n2k) → \mathrm{SpGr}(2n,2k), p ↦ π(p) = pp^+.
+```
+
+The tangent space is either the tangent space from the symplectic Stiefel manifold, where
+tangent vectors are representers of their corresponding congruence classes, or for the
+representation as projectors, using a [`ProjectorTVector`](@ref) as
+
+```math
+  T_p\operatorname{SpGr}(2n, 2k, ℝ) =
+  \bigl\{ [X,p] \ \mid\ X ∈ \mathfrak{sp}(2n,ℝ), Xp+pX = X \bigr\},
+```
+
+where ``[X,p] = Xp-pX`` denotes the matrix commutator and
+``\mathfrak{sp}(2n,ℝ)`` is the Lie algebra of the symplectic group consisting of [`HamiltonianMatrices`](@ref).
```

Comment:
I don't think we actually have the symplectic group as a group in Manifolds.jl? Maybe we should have an issue open about it.

Reply:
We at least have the manifold that would be based on, HamiltonianMatrices, now, but sure, we could open an issue about that (but I will not add it here, this PR is already about a month longer than I thought).

Reply:
Sure, this PR is definitely not a good place for even more new stuff.

Reply:
I've opened an issue: #705 .

Reply:
Oh, no! I had hoped at some point to reduce our open issues, my feeling is every PR to close one opens 2 new ones ;)

Reply:
Unfortunately, it's much easier to open new issues than to close existing ones, so here we are :sweat_smile: